src/sentry/eventstore/models.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import absolute_import
+
+from datetime import datetime
+from django.utils import timezone
+
+from sentry.models import EventCommon, EventDict
+from sentry.db.models import NodeData
+
+
+def ref_func(x):
+    return x.project_id or x.project.id
+
+
+class Event(EventCommon):
+    def __init__(self, project_id, event_id, group_id=None, data=None):
+        self.project_id = project_id
+        self.event_id = event_id
+        self.group_id = group_id
+        self.data = data
+        super(Event, self).__init__()
+
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        # do not pickle cached info.  We want to fetch this on demand
+        # again.  In particular if we were to pickle interfaces we would
+        # pickle a CanonicalKeyView which old sentry workers do not know
+        # about
+        state.pop("_project_cache", None)
+        state.pop("_environment_cache", None)
+        state.pop("_group_cache", None)
+        state.pop("interfaces", None)
+
+        return state
+
+    @property
+    def data(self):
+        return self._data
+
+    @data.setter
+    def data(self, value):
+        node_id = Event.generate_node_id(self.project_id, self.event_id)
+        self._data = NodeData(
+            node_id, data=value, wrapper=EventDict, ref_version=2, ref_func=ref_func
+        )
+
+    @property
+    def platform(self):
+        return self.data.get("platform", None)
+
+    @property
+    def datetime(self):
+        recorded_timestamp = self.data.get("timestamp")
+        date = datetime.fromtimestamp(recorded_timestamp)
+        date = date.replace(tzinfo=timezone.utc)
+        return date
+
+    def save(self):
+        """
+        Saves event to nodestore. Should also insert event to Eventstream in future.
+        """
+        self._data.save()
```

Comment:
This is not a minor decision. Deciding whether to add write methods on the event interface we provide through eventstore has a lot of ramifications to take into account. 
It may be a viable pattern (like the one provided by the Django ORM) but there is a major difference there: all writes are synchronous, here it would be more complex.
1) providing a save method means it should always work. How would this work for updates? Would you put a replacement on eventstream ?
2) Calling the method `save` seems to imply you are giving a read your write guarantee, which is not exactly correct, if someone reloads the event from snuba there is no guarantee. If we add this method we should rename it to something that does not promise any guarantee
3) If we add a save method on the event class it means we should always save events this way. What about the initial creation of the event? Are we going to create an instance through eventstore and call save on the object itself ?

Reply:
I was only thinking of the initial creation case here, apart from that we only rewrite event data under a very limited set of circumstances (merge, unmerge and delete).

In all of those 3 special cases, it doesn't make sense to go through eventstore since we are never operating on a single event - we are passsing a list of group IDs to snuba via evenstream to be merged, unmerged or deleted (and we don't actually have the events or even the event IDs at that point).

Since this method isn't being used (except in the tests) I've removed it for now.